File: test-suite-python/src/test/python/micronaut/docs/aws/sdk/v1/AmazonS3Factory.py
# tag::clazz[]
from com.amazonaws.services.s3 import AmazonS3, AmazonS3ClientBuilder
from jakarta.inject import Singleton
from micronaut.aws.sdk.v1 import EnvironmentAWSCredentialsProvider
from micronaut.context.annotation import Bean, Factory
from micronaut.context.env import Environment


@Factory
class AmazonS3Factory:

    @Singleton
    @Bean(preDestroy="shutdown")
    def amazon_s3(self, environment: Environment) -> AmazonS3:
        amazon_s3_client_builder = AmazonS3ClientBuilder.standard()
        amazon_s3_client_builder.setCredentials(EnvironmentAWSCredentialsProvider(environment))  # <1>
        return amazon_s3_client_builder.build()
# end::clazz[]

File: test-suite/src/test/java/io/micronaut/docs/aws/sdk/v1/AmazonS3Factory.java
package io.micronaut.docs.aws.sdk.v1;

//tag::clazz[]
import com.amazonaws.services.s3.AmazonS3;
import com.amazonaws.services.s3.AmazonS3ClientBuilder;
import io.micronaut.aws.sdk.v1.EnvironmentAWSCredentialsProvider;
import io.micronaut.context.annotation.Bean;
import io.micronaut.context.annotation.Factory;
import io.micronaut.context.env.Environment;
import jakarta.inject.Singleton;

@Factory
public class AmazonS3Factory {

    @Singleton
    @Bean(preDestroy = "shutdown")
    public AmazonS3 amazonS3(Environment environment) {
        AmazonS3ClientBuilder amazonS3ClientBuilder = AmazonS3ClientBuilder.standard();
        amazonS3ClientBuilder.setCredentials(new EnvironmentAWSCredentialsProvider(environment)); // <1>
        return amazonS3ClientBuilder.build();
    }
}
//end::clazz[]
